packages/core/src/rules/ajv.ts

```diff
@@ -39,11 +39,15 @@ function getAjvValidator(
   disallowAdditionalProperties: boolean,
 ): ValidateFunction | undefined {
   const ajv = getAjv(resolve, disallowAdditionalProperties);
-
+  if (!ajv.opts.defaultAdditionalProperties !== disallowAdditionalProperties) {
+    if (isRef(schema)) {
+      schema = resolve(schema);
+    }
+    schema.additionalProperties = schema.additionalProperties ?? !disallowAdditionalProperties;
```

Comment:
Ah, I just realized. It won't work on nested levels. Only on top level.

e.g.:

```
type: object
properties:
  a:
    allOf:
      - ...
```

We need to change the approach I think somehow...

I think we may need to do some changes in our fork of AJV validator iteself: https://github.com/Redocly/ajv

Reply:
Looks to me like you are trying force `additionalProperties` to behave just like `unevaluatedProperties`. Wouldn't it be better to have a flag disallowUnevaluatedProperties and overwrite unevaluatedProperties with that (at least in openapi 3.1)?

Reply:
Hmm. I didn't know about `unevaluatedProperties`. We'll check it out and do appropriate changes. Thanks!